docs/3.0rc/resources/upgrade-prefect-3.mdx

```diff
@@ -0,0 +1,233 @@
+This means that some patterns supported by Prefect 2 are no longer supported with Prefect 3, or are enabled differently.
+This guide covers changes that you may need to make to ensure your Prefect 2 workflows keep flowing.
+
+<Info>
+Prefect 3 is an upgrade to the open source `prefect` Python package
+The Prefect 3 release does not involve any changes to Prefect Cloud.
+All previous and future Prefect 2 releases will continue to work with Prefect Cloud.
+</Info>
+
+## Module locations and names
+
+Some modules have been renamed, reorganized, or removed for clarity. 
+Objects imported from changed modules will continue to work until their six month deprecation period ends, but will raise deprecation warnings. 
+See the [migration script](https://github.com/PrefectHQ/prefect/blob/f767724ab28278446456b024b5c1c73f76ebcbb5/src/prefect/_internal/compatibility/migration.py#L22) for the full list of modules that have been moved and removed.
+
+## Asynchronous task execution behavior
+
+Prefect 2 abstracts away certain task execution concerns, which is convenient in some cases, but comes at the expense of unexpected behavior in other cases. 
+Prefect 3 defaults to running tasks as they would run as normal Python code, and requires you to be more explicit when you would like to modify task execution behavior. 
+This results in clearer behavior with fewer surprises.
+
+In Prefect 3, all workflow code, including task functions, runs on the main thread. This means:
+
+- Python code that is not explicitly defined as asynchronous will execute synchronously, as normal Python code does.
+- Non-thread safe objects, such as database connections, can be shared globally within a workflow.
+
+To execute tasks asynchronously, you must either write asynchronous Python functions or use the `Task.submit` method to submit to a Prefect task runner.
+Prefect 3 encourages conventional asynchronous Python programing principles. 
+Updating your workflows to bring them in line with these principles requires code changes. 
+Here are some examples:
+
+**Synchronous tasks and flows cannot be called from asynchronous tasks or flows as they could in Prefect 2.**
+
+<CodeGroup>
+```python Prefect 2
+@task
+async def my_task():
+    ...
+
+
+# Enclosing flow can be sync
+@flow
+def my_flow():
+    return my_task()
+
+    
+if __name__ == "__main__":
+    my_flow()
+```
+
+```python Prefect 3
+@task
+async def my_task():
+    ...
+
+
+# Enclosing flow must be async as well
+@flow
+async def my_flow():
+    return await my_task()
+
+
+if __name__ == "__main__":
+    asyncio.run(my_flow())
+``` 
+</CodeGroup>
+
+Because `Task.submit` is always a synchronous call, it does not need to be awaited. 
+`Task.submit` is still non-blocking and execution of the submitted task is also still non-blocking. 
+So, *you do not need to use the `await` keyword in front of your `.submit` call*:
```

Comment:
```suggestion
*You do not need to use the `await` keyword in front of your `.submit` call* because `Task.submit` is always a synchronous call. `Task.submit` and the execution of the submitted task remain non-blocking:
```

I'm not totally sure this is better, I just felt the original could be rephrased/more concise.

Reply:
Blocking vs. non-blocking feels like a trap. 😂 "Non-blocking" in modern Python usually refers to _non-blocking IO_, which is asynchronous Python code that uses `await`, etc. In that sense, `submit` is a blocking method call that makes a blocking-IO API request to schedule your task to run. Then your task runs asynchronously.